src/seedu/addressbook/data/person/Name.java

```
@@ -51,6 +51,9 @@ public static boolean isValidName(String test) {
 	 * other
 	 */
 	public boolean isSimilar(Name other) {
+		if(other == null) {
+			return false;
```

Should throw an exception instead of returning false.